app/controllers/application_controller.rb

```diff
@@ -46,7 +46,7 @@ def set_meta_images_for_letter(first_letter)
   end
 
   def redirect_to_authenticated_settings(author)
-    redirect_to "#{ENV['HOST']}/authors/#{author.id}/settings/?secret=#{author.secret}", :status => 303
+    redirect_to "#{author.url}/authors/#{author.id}/settings/?secret=#{author.secret}", :status => 303
```

Comment:
Shouldn't this url be "#{author.url}/settings/?secret=#{author.secret}"?

Reply:
Nope, the settings url is /authors/settings :)

Reply:
author.url already maps to authors/[id] (or /@username or custom domain). This url for an author without a username will map to authors/[id]/authors/[id]

Reply:
Perhaps use author.get_host instead of author.url?

Reply:
nice catch! @antsgar , fixed!